### ISSUE TITLE
Problem of headless EGL rendering?

### ISSUE THREAD

**Issue by wangg12** [vispy/vispy#1718]
I was trying to run `python examples/offscreen/simple_egl.py` remotely via ssh, but got this error:
```
RuntimeError: Could not import backend "EGL":
Could not create display
```
Is there any problem with the EGL backend for remotely rendering?

**Comment by djhoese** [comment]
Disclaimer: I have no real experience with the EGL backend or headless rendering.

There were some updates recently to the EGL backend by @yogi81 in #1602 which was eventually merged as part of #1637. This was discussed a little bit more recently here: https://github.com/vispy/vispy/issues/1684.  Bottom line is that I'm not sure the EGL backend completely works headless and may require more work to do so. It was something that @yogi81 had mentioned in the various EGL issues and pull requests.

That said, are you doing X forwarding when connecting over SSH?

**Comment by wangg12** [comment]
Actually it should be possible to run without any X forwarding or local display with EGL backend as implemented in pyrender [https://github.com/mmatl/pyrender/issues/7](https://github.com/mmatl/pyrender/issues/7). Maybe currently the implementation in vispy can not acheive that.

**Comment by djhoese** [comment]
Yes, that's what I was trying to say. The "EGL backend" (meaning in vispy) may not be capable of this at this moment. It is not something I have tested. Any help/information/testing/updates you can provide to help figure this out would be greatly appreciated. Proving that pyrender works for your setup would be a good starting point.

**Comment by wangg12** [comment]
Yes, pyrender works for my setup as I said in the aforementioned issue.

**Comment by djhoese** [comment]
Ah didn't realize you commented. Sorry.

**Comment by djhoese** [comment]
Lol. Glad you got it to work. Thanks for closing the issue.
